README.md

```diff
@@ -1,84 +1,143 @@
-## blocklock-solidity
+# blocklock-solidity
 
-This repository contains the Solidity-based smart contracts library that facilitates Randamu's on-chain timelock encryption and decryption.
+[![Solidity ^0.8.x](https://img.shields.io/badge/Solidity-%5E0.8.x-blue)](https://soliditylang.org/)
+[![License: MIT](https://img.shields.io/badge/License-MIT-green)](LICENSE)
+[![Foundry Tests](https://img.shields.io/badge/Tested%20with-Foundry-red)](https://book.getfoundry.sh/)
 
-By leveraging this library, developers can implement time-based data unlocking mechanisms securely in their smart contracts.
+A Solidity library enabling on-chain blocklock encryption and decryption, from the [dcipher threshold network](https://dcipher.network/). This facilites secure time-based data unlocking mechanisms for smart contracts.
 
-This library is designed with modularity and simplicity in mind, allowing developers to extend and integrate it into their existing projects easily.
+## ✨ Overview
+
+Controlling access to data based on time is crucial for various use cases, such as auctions, voting, and content release schedules. `blocklock-solidity` provides developers with tools to implement blocklock encryption on-chain, ensuring that encrypted data can only be decrypted after a specified block height, thus enhancing security and fairness in time-sensitive operations. 
```

Comment:
We previously discussed being consistent in the docs and tutorials / blogs, in term of using timelock encryption or blocklock encryption. 
In this context, timelock encryption might be better for consistency.

Reply:
That's fine for now, thanks

Reply:
I will update the readme to use timelock encryption. But keep in mind, this library is called blocklock.solidity which is slightly not consistent.